tutorials/developer-tips.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Developer tips
+subtitle: 
+published: 
+code: 
+difficulty: 4
+---
+
+# Expected tokens #
+When you get a [`ParserError`](https://www.stackage.org/haddock/lts-7.18/megaparsec-5.0.1/Text-Megaparsec.html#t:ParseError), you can use the `errorExpected` function together with the [`toList`](https://www.stackage.org/haddock/lts-7.18/containers-0.5.7.1/Data-Set.html#v:toList) to get a list of expected [`ErrorItem`'s](https://www.stackage.org/haddock/lts-7.18/megaparsec-5.0.1/Text-Megaparsec.html#t:ErrorItem). These `ErrorItem`'s can be one of the following:
```

Comment:
Sorry, this is sort-of obivous. Even when parse error is displayed, it's rendered with “or” between expected tokens. How can several expected tokens match at the same time anyway?